iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -298,14 +286,56 @@ inline void optional<T>::destruct_value() noexcept
     value().~T();
     m_hasValue = false;
 }
-
+// AXIVION Next Construct AutosarC++19_03-M17.0.3 : make_optional is defined within iox::cxx and
+// is not overridden
 template <typename OptionalBaseType, typename... Targs>
 inline optional<OptionalBaseType> make_optional(Targs&&... args) noexcept
 {
-    optional<OptionalBaseType> returnValue = nullopt_t();
+    optional<OptionalBaseType> returnValue{nullopt_t()};
     returnValue.emplace(std::forward<Targs>(args)...);
     return returnValue;
 }
+
+template <typename T>
+bool operator==(const optional<T>& lhs, const optional<T>& rhs) noexcept
+{
+    auto bothNull = !lhs.has_value() && !rhs.has_value();
+    auto bothEqual = (lhs.has_value() && rhs.has_value()) && (*lhs == *rhs);
```

Comment:
```suggestion
    const auto bothValuesEqual = (lhs.has_value() && rhs.has_value()) && (*lhs == *rhs);
```

Alternatively a one liner but the compiler should optimize it anyway and this is "intentional programming".
Now the question for optimization nerds (as myself) is whether it still is able to benefit from shortcircuiting. I would think so, with optimization, but one would need to fire up godbolt to confirm.

Note I am fine either way. I just think some analysis will complain if the helper expressions are not `const`. I consider this unnecessary for such a simple functions but the tools do not.

TLDR: either directly return the expanded expression or add const as suggested.

Reply:
this is a good suggestion. thanks.